bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/discovery/EpsonProjectorDiscoveryParticipant.java

```diff
@@ -0,0 +1,86 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.epsonprojector.internal.discovery;
+
+import static org.openhab.binding.epsonprojector.internal.EpsonProjectorBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.sddp.SddpDevice;
+import org.openhab.core.config.discovery.sddp.SddpDiscoveryParticipant;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Discovery Service for Epson Projectors that support SDDP.
+ *
+ * @author Michael Lobstein - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class EpsonProjectorDiscoveryParticipant implements SddpDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorDiscoveryParticipant.class);
+
+    private static final String EPSON = "EPSON";
+    private static final String TYPE_PROJECTOR = "PROJECTOR";
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(SddpDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(3);
+            final String label = device.manufacturer + " " + device.model;
+
+            properties.put(Thing.PROPERTY_MAC_ADDRESS, uid.getId());
+            properties.put(THING_PROPERTY_HOST, device.ipAddress);
+            properties.put(THING_PROPERTY_PORT, DEFAULT_PORT);
+
+            final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS).withLabel(label).build();
+
+            logger.debug("Created a DiscoveryResult for device '{}' with UID '{}'", label, uid.getId());
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(SddpDevice device) {
+        if (device.manufacturer.toUpperCase(Locale.ENGLISH).contains(EPSON)
+                && device.type.toUpperCase(Locale.ENGLISH).contains(TYPE_PROJECTOR) && !device.macAddress.isBlank()) {
+
+            logger.debug("Epson projector with mac {} found at {}", device.macAddress, device.ipAddress);
+            return new ThingUID(THING_TYPE_PROJECTOR_TCP, device.macAddress);
```

Comment:
A mac address contains ":" and I doubt such character are valid for a thing id.

Reply:
We discussed this in the core PR. You are correct that ':' is not accepted in a thing UID. There is no formal standard for the format of Mac addresses -- they can be delimited with either ':' or '-', and they can use either upper or lower case hex characters e.g. `E0:DA:DC:15:28:02` resp. `e0-da-dc-15-28-02` => Therefore in SddpDevice the mac address uses the dash delimited lower case format.